phoenix-core-client/src/main/java/org/apache/phoenix/query/QueryServices.java

```diff
@@ -486,6 +486,8 @@ public interface QueryServices extends SQLCloseable {
 
     int DEFAULT_PHOENIX_GET_REGIONS_RETRIES = 10;
 
+    String PHOENIX_GET_METADATA_READ_LOCK_ENABLED = "phoenix.get.metadata.read.lock.enabled";
```

Comment:
Whats the risk that requires this be parameterized?  Also when did the previous change with exclusive writelock go ou?

Reply:
Just the ability to quickly to disable this change in production if it causes any issues.

Reply:
Is it not enabled prior to this release?

Reply:
> Also when did the previous change with exclusive write lock go out?
@dbwong This has always been the behaviour in production. We want to change it to using read locks for getTable RPC calls.